### PR TITLE
Adding Simple RecyclerView Sample

### PR DESCRIPTION
**Pull request by MagicalMeghan**
Hello - adding in a very simple RecyclerView example

### CODE REVIEW

File: RecyclerViewSimple/app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,39 @@
+apply plugin: 'com.android.application'
```
Comment (chrisbanes) [inline RecyclerViewSimple/app/build.gradle:1]: Missing copyright header

File: RecyclerViewSimple/app/src/main/java/com/example/recyclersample/Datasource.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2020 Google Inc. All rights reserved.
```
Comment (chrisbanes) [inline RecyclerViewSimple/app/src/main/java/com/example/recyclersample/Datasource.kt:2]: This should be the AOSP copyright (ditto other files)

File: RecyclerViewSimple/app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (chrisbanes) [inline RecyclerViewSimple/app/src/main/AndroidManifest.xml:1]: Missing copyright header

File: RecyclerViewSimple/app/src/androidTest/java/com/example/recyclersample/ExampleInstrumentedTest.kt [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package com.example.recyclersample
```
Comment (chrisbanes) [inline RecyclerViewSimple/app/src/androidTest/java/com/example/recyclersample/ExampleInstrumentedTest.kt:1]: Missing copyright header

File: RecyclerViewSimple/app/src/androidTest/java/com/example/recyclersample/ExampleInstrumentedTest.kt [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.ext.junit.runners.AndroidJUnit4
+
+import org.junit.Test
+import org.junit.runner.RunWith
+
+import org.junit.Assert.*
```
Comment (chrisbanes) [inline RecyclerViewSimple/app/src/androidTest/java/com/example/recyclersample/ExampleInstrumentedTest.kt:9]: Wildcard imports are discouraged.

File: RecyclerViewSimple/app/src/main/java/com/example/recyclersample/Datasource.kt [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.recyclersample
+
+import android.content.Context
+
+class Datasource(val context: Context) {
```
Comment (chrisbanes) [inline RecyclerViewSimple/app/src/main/java/com/example/recyclersample/Datasource.kt:21]: private val?

File: RecyclerViewSimple/app/src/main/java/com/example/recyclersample/FlowerAdapter.kt [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+package com.example.recyclersample
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+
+class FlowerAdapter(val flowerList: Array<String>) : RecyclerView.Adapter<FlowerAdapter.FlowerViewHolder>() {
```
Comment (chrisbanes) [inline RecyclerViewSimple/app/src/main/java/com/example/recyclersample/FlowerAdapter.kt:25]: Line looks too long (max 100 chars)

File: RecyclerViewSimple/build.gradle [hunk 8]
```diff
@@ -0,0 +1,29 @@
+// Top-level build file where you can add configuration options common to all sub-projects/modules.
+
+buildscript {
+    ext.kotlin_version = '1.3.72'
```
Comment (chrisbanes) [inline RecyclerViewSimple/build.gradle:4]: 1.4

File: RecyclerViewSimple/build.gradle [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    repositories {
+        google()
+        jcenter()
+        
+    }
+    dependencies {
+        classpath 'com.android.tools.build:gradle:3.6.3'
```
Comment (chrisbanes) [inline RecyclerViewSimple/build.gradle:11]: 4.0.1

File: RecyclerViewSimple/gradle.properties [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+# When configured, Gradle will run in incubating parallel mode.
+# This option should only be used with decoupled projects. More details, visit
+# http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
+# org.gradle.parallel=true
+# AndroidX package structure to make it clearer which packages are bundled with the
+# Android operating system, and which are packaged with your app's APK
+# https://developer.android.com/topic/libraries/support-library/androidx-rn
+android.useAndroidX=true
+# Automatically convert third-party libraries to use AndroidX
+android.enableJetifier=true
```
Comment (chrisbanes) [inline RecyclerViewSimple/gradle.properties:19]: Shouldn't need jetifier

File: RecyclerViewSimple/gradle/wrapper/gradle-wrapper.properties [hunk 11]
```diff
@@ -0,0 +1,6 @@
+#Fri Jul 17 12:26:11 PDT 2020
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-all.zip
```
Comment (chrisbanes) [inline RecyclerViewSimple/gradle/wrapper/gradle-wrapper.properties:6]: 6.6

**Review by chrisbanes — COMMENTED** [review]
Took a quick pass through. Generally:

- Needs copyright headers on all source files
- Run ktlint over all kotlin files, there's a few code style nits
- Needs inline comments explaining the sample

File: RecyclerViewSimple/gradle.properties (hunk 10 above)
Comment (MagicalMeghan) [reply RecyclerViewSimple/gradle.properties:19]: This came standard with an empty project

File: RecyclerViewSimple/gradle/wrapper/gradle-wrapper.properties (hunk 11 above)
Comment (MagicalMeghan) [reply RecyclerViewSimple/gradle/wrapper/gradle-wrapper.properties:6]: Standard with empty project
gradle-5.6.6-all.zip? This gave me an error

**Comment by MagicalMeghan** [comment]
Added some changes in @chrisbanes !

File: RecyclerViewSimple/app/build.gradle [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+apply plugin: 'com.android.application'
+apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-android-extensions'
+
+android {
+    compileSdkVersion 29
```
Comment (chrisbanes) [inline RecyclerViewSimple/app/build.gradle:22]: 30

File: RecyclerViewSimple/app/build.gradle [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+    buildToolsVersion "29.0.2"
```
Comment (chrisbanes) [inline RecyclerViewSimple/app/build.gradle:23]: Don't need this line

File: RecyclerViewSimple/app/build.gradle [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+
+    defaultConfig {
+        applicationId "com.example.recyclersample"
+        minSdkVersion 21
+        targetSdkVersion 29
```
Comment (chrisbanes) [inline RecyclerViewSimple/app/build.gradle:28]: 30

File: RecyclerViewSimple/app/build.gradle [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+        }
+    }
+
+}
+
+dependencies {
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
```
Comment (chrisbanes) [inline RecyclerViewSimple/app/build.gradle:45]: Can remove this line

File: RecyclerViewSimple/app/build.gradle [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation 'androidx.appcompat:appcompat:1.1.0'
+    implementation 'androidx.core:core-ktx:1.3.0'
+    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
```
Comment (chrisbanes) [inline RecyclerViewSimple/app/build.gradle:49]: 2.0.1 is latest

File: RecyclerViewSimple/app/proguard-rules.pro [hunk 17]
```diff
@@ -0,0 +1,21 @@
+# Add project specific ProGuard rules here.
```
Comment (chrisbanes) [inline RecyclerViewSimple/app/proguard-rules.pro:1]: Remove this file if you're not minifying

File: RecyclerViewSimple/app/build.gradle [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
+        versionCode 1
+        versionName "1.0"
+
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+    }
+
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
```
Comment (chrisbanes) [inline RecyclerViewSimple/app/build.gradle:38]: Remove this line if minify is disabled

File: RecyclerViewSimple/build.gradle [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+        // NOTE: Do not place your application dependencies here; they belong
+        // in the individual module build.gradle files
+    }
+}
+
+allprojects {
+    repositories {
+        google()
+        jcenter()
+
```
Comment (chrisbanes) [inline RecyclerViewSimple/build.gradle:37]: nit: empty line

File: RecyclerViewSimple/gradle.properties [hunk 20]
```diff
@@ -0,0 +1,21 @@
...
+org.gradle.jvmargs=-Xmx1536m
+# When configured, Gradle will run in incubating parallel mode.
+# This option should only be used with decoupled projects. More details, visit
+# http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
+# org.gradle.parallel=true
+# AndroidX package structure to make it clearer which packages are bundled with the
+# Android operating system, and which are packaged with your app's APK
+# https://developer.android.com/topic/libraries/support-library/androidx-rn
+android.useAndroidX=true
+# Automatically convert third-party libraries to use AndroidX
```
Comment (chrisbanes) [inline RecyclerViewSimple/gradle.properties:18]: Can just remove these lines

File: RecyclerViewSimple/build.gradle [hunk 21]
```diff
@@ -0,0 +1,43 @@
...
+allprojects {
+    repositories {
+        google()
+        jcenter()
+        
+    }
+}
+
+task clean(type: Delete) {
```
Comment (chrisbanes) [inline RecyclerViewSimple/build.gradle:41]: Can remove this task. No longer needed

**Review by chrisbanes — COMMENTED** [review]
LGTM. Just a few nits.

**Pull request merged**